### PR TITLE
[FIX]: create_task response data

### LINKED ISSUE #207: Adding useful reply data to `create_task`
Aim: Increase usability of `create_task` msg

(This might be useful for other messages)

At the moment, to use cw-croncat to `create_task` we rely on the reply attributes in the wasm event to get the `task_hash` to [update our state](https://github.com/nymlab/vectis-plugins/blob/8215707eee771f8dccad5ded3f688fb8bd5e8b6c/cronkitty/src/lib.rs#L53). 

Perhaps it can be return in the `data` field of the `Response` so it can be easily parsed.

### PR DESCRIPTION
**Pull request by deveusss**
closes #207

### CODE REVIEW

File: contracts/cw-croncat/src/tasks.rs [hunk 1]
```diff
@@ -411,6 +411,7 @@ impl<'a> CwCroncat<'a> {
         };
 
         Ok(Response::new()
+            .set_data(format!("task_hash {}", hash).into_bytes())
```
Comment (Buckram123) [inline contracts/cw-croncat/src/tasks.rs:414]:
```suggestion
            .set_data(hash.as_bytes())
```
I think should work

**Review by Buckram123 — CHANGES_REQUESTED** [review]
small fix

File: contracts/cw-croncat/src/tasks.rs (hunk 1 above)
Comment (Buckram123) [reply contracts/cw-croncat/src/tasks.rs:409]: Oops, just noticed that there is also "task_hash" string as well) Anyway do we want to format it in some different way?

File: contracts/cw-croncat/src/contract.rs [hunk 2]
```diff
@@ -201,7 +201,6 @@ impl<'a> CwCroncat<'a> {
                 to_binary(&self.query_get_tasks_by_owner(deps, owner_id)?)
             }
             QueryMsg::GetTask { task_hash } => to_binary(&self.query_get_task(deps, task_hash)?),
-            QueryMsg::GetTaskHash { task } => to_binary(&self.query_get_task_hash(*task)?),
```
Comment (TrevorJTClarke) [inline contracts/cw-croncat/src/contract.rs:204]: Hey @deveusss, im not happy with removing this. Do you see duplicated way of getting task hash? External integrations (web, etc) should be able to compute tash hash with query, so we have a reliable way without additional libs or self-compute that could be innacurate.
If this isn't duplicated, we need to keep this as the website uses it.

**Review by TrevorJTClarke — CHANGES_REQUESTED** [review]

File: contracts/cw-croncat/src/contract.rs (hunk 2 above)
Comment (Buckram123) [reply contracts/cw-croncat/src/contract.rs:204]: @TrevorJTClarke I think it's still supposed to be removed in #209 right?

Comment (TrevorJTClarke) [reply contracts/cw-croncat/src/contract.rs:204]: I dont think so - adding simulate is different use case thank just purely getting task_hash. 
Having tash_hash part of simulate is def a nice thing!

Comment (deveusss) [reply contracts/cw-croncat/src/contract.rs:204]: brought GetTaskHash back

File: contracts/cw-croncat/src/tasks.rs (hunk 1 above)
Comment (mikedotexe) [reply contracts/cw-croncat/src/tasks.rs:414]: Curious why we're not using `.add_attribute` here, @deveusss?

Comment (deveusss) [reply contracts/cw-croncat/src/tasks.rs:414]: @mikedotexe  image is not loading)

Comment (deveusss) [reply contracts/cw-croncat/src/tasks.rs:414]: @mikedotexe The reason is the consumer, like vectis, could get data from events

**Review by mikedotexe — APPROVED** [review]

**Review by TrevorJTClarke — APPROVED** [review]

File: contracts/cw-croncat/src/tasks.rs (hunk 1 above)
Comment (TrevorJTClarke) [reply contracts/cw-croncat/src/tasks.rs:414]: @deveusss i still see format! in there, can you change to only hash? don't want vectis or others having to string split `task_hash jfdiksajfkldsjfdkslajf`

**Review by TrevorJTClarke — CHANGES_REQUESTED** [review]

File: contracts/cw-croncat/src/tasks.rs (hunk 1 above)
Comment (deveusss) [reply contracts/cw-croncat/src/tasks.rs]: thank you, done)

**Review by TrevorJTClarke — APPROVED** [review]

**Pull request merged**